bika/lims/workflow/analysisrequest/events.py

```diff
@@ -41,6 +42,22 @@ def before_sample(analysis_request):
         analysis_request.setSampler(api.get_current_user().id)
 
 
+def after_no_sampling_workflow(analysis_request):
+    """Function triggered after "no_sampling_workflow transition for the
+    Analysis Request passed in is performed
+    """
+    setup = api.get_setup()
+    if setup.getAutoreceiveSamples():
+        # Auto-receive samples is enabled. Receive the sample automatically,
+        # but only if the current user is a laboratory contact
+        user = api.get_current_user()
+        contact = api.get_user_contact(user, contact_types=["LabContact"])
```

Comment:
Wouldn't it be better here to check the role of the user? E.g. if the user has the permission to receive the sample?

Reply:
OMG, you're totally right!. Thanks for pointing this out.

Reply:
Also consider to cache the user role in the request in case of multi-reception. Not sure how "expensive" the role lookup is in terms of performance...